src/pages/user/review/[username].tsx

```diff
@@ -1,57 +1,223 @@
+import { useRouter } from "next/router";
+import { getTrainerReviews, ReviewDetail } from "@/services/trainer.service";
+import {StarIcon} from "@heroicons/react/24/outline";
 
 const Review = () => {
-  const ReviewBox = () => {
+  const [showModal, setShowModal] = useState(false);
+
+  const [trainerProfile, setTrainerProfile] = useState<UserProfile>({
+    username: "user",
+    firstname: "firstname",
+    lastname: "lastname",
+    birthdate: new Date(),
+    citizenId: "none",
+    gender: "none",
+    phoneNumber: "none",
+    address: "none",
+    usertype: "none",
+    lat: 0,
+    lng: 0,
+  });
+
+  
+
+  const [trainerReviews, setTrainerReviews] = useState<ReviewDetail[]>([]);
+
+  const router = useRouter();
+  const { username } = router.query;
+  const [loading, setLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+  if (typeof username == "string") {
+    getTrainerProfile(username)
+      .then((res1) => {
+        setLoading(true);
+        setTrainerProfile(res1.userProfile);
+        setLoading(false);
+      })
+      .catch(() => router.back())
+    getTrainerReviews(username)
+      .then((res2) => {
+        setLoading(true);
+        setTrainerReviews(res2.reviewList);
+        setLoading(false);
+      })
+    .catch(() => router.back())
+  } else {
+    router.back();
+  }
+}, [username]);
+
+  const Image = () => {
     return (
-      <div className="h-[20%] w-[80%] bg-white mb-[5%] border-2 border-gray rounded-3xl items-center justify-between drop-shadow-lg hover:bg-gray-light"></div>
+      <div className="max-h-min overflow-hidden mt-6 md:mt-10 flex justify-center">
+        <img
+          src="/default_profile.jpg"
+          alt=""
+          className="rounded-2xl object-contain w-28 h-28 sm:w-36 sm:h-36 md:w-fit md:h-fit"
+        />
+      </div>
     );
   };
+  const StarRating = (rating:{ rating: number; }) => {
+    return(
+      <div className="flex flex-row">
+        {[...Array<Number>(Math.round(rating.rating)),].map((e, idx) => {
+          return (
+            <StarIcon
+              key={idx}
+              className="h-6 w-6 fill-yellow stroke-yellow"
+              strokeWidth={2}
+            ></StarIcon>
+          );
+        })}
+        {[...Array<Number>(5 -Math.round(rating.rating)),].map((e, idx) => {
+          return (
+            <StarIcon
+              key={idx}
+              className="h-6 w-6 fill-none stroke-yellow"
+              strokeWidth={2}
+            ></StarIcon>
+          );
+          })}
+      </div>
+    )
+  }
 
-  return (
-    <main className="w-full h-screen pt-20 pl-6 flex flex-col bg-backgroundColor">
-      <div className="flex items-center mb-3 md:mb-8">
-        <BackButton href="/user/booking" />
-        <p className="text-xl md:text-3xl mx-5 md:mx-10 font-bold display: inline-flex">
-          Rating and Review
-        </p>
+  const EmptyStar = () => {
+    return (
+      <div>
+        <svg
```

Comment:
you should use `<StarIcon classname='.....' />` form heroicons instead of this long svg